Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/Events.proto

```diff
@@ -22,31 +34,845 @@ message EventTypeQuery{
+    ReadWrite = 3; 
+    // Read | Write | Probe | Enumerate | EnumerationProbe
+    All = 31;
+}
+
+enum ReportedFileOperation_XLGpp{
+    Unknown = 0;
+    CreateFile = 1;
+    GetFileAttributes = 2;
+    GetFileAttributesEx = 3;
+    Process = 4;
+    FindFirstFileEx = 5;
+    FindNextFile = 6;
+    CreateDirectory = 7;
+    DeleteFile = 8;
+    MoveFileSource = 9;
+    MoveFileDestination = 10;
+    SetFileInformationByHandleSource = 11;
+    SetFileInformationByHandleDest = 12;
+    ZwSetRenameInformationFileSource = 13;
+    ZwSetRenameInformationFileDest = 14;
+    ZwSetLinkInformationFile = 15;
+    ZwSetDispositionInformationFile = 16;
+    ZwSetModeInformationFile = 17;
+    ZwSetFileNameInformationFileSource = 18;
+    ZwSetFileNameInformationFileDest = 19;
+    CopyFileSource = 20;
+    CopyFileDestination = 21;
+    CreateHardLinkSource = 22;
+    CreateHardLinkDestination = 23;
+    RemoveDirectory = 24;
+    NtQueryDirectoryFile = 25;
+    ZwQueryDirectoryFile = 26;
+    NtCreateFile = 27;
+    ZwCreateFile = 28;
+    ZwOpenFile = 29;
+    ChangedReadWriteToReadAccess = 30;
+    FirstAllowWriteCheckInProcess = 31;
+    ReparsePointTarget = 32;
+    CreateSymbolicLinkSource = 33;
+    MoveFileWithProgressSource = 34;
+    MoveFileWithProgressDest = 35;
+    MultipleOperations = 36;
+    ProcessExit = 37;
+    MacLookup = 38;
+    MacReadlink = 39;
+    MacVNodeCreate = 40;
+    KAuthMoveSource = 41;
+    KAuthMoveDest = 42;
+    KAuthCreateHardlinkSource = 43;
+    KAuthCreateHardlinkDest = 44;
+    KAuthCopySource = 45;
+    KAuthCopyDest = 46;
+    KAuthDeleteDir = 47;
+    KAuthDeleteFile = 48;
+    KAuthOpenDir = 49;
+    KAuthReadFile = 50;
+    KAuthCreateDir = 51;
+    KAuthWriteFile = 52;
+    KAuthClose = 53;
+    KAuthCloseModified = 54;
+    KAuthGetAttributes = 55;
+    KAuthVNodeExecute = 56;
+    KAuthVNodeWrite = 57;
+    KAuthVNodeRead = 58;
+    KAuthVNodeProb = 59;
+}
+
+enum FingerprintComputationKind_XLGpp{
+    Execution = 0;
+    CacheCheck = 1;
+}
+
+enum ObservedInputType_XLGpp{
+    AbsentPathProbe = 0;
+    FileContentRead = 1;
+    DirectoryEnumeration = 2;
+    ExistingDirectoryProbe = 3;
+    ExistingFileProbe = 4;
+}
+
+enum PreserveOutputsMode_XLGpp{
+    Disabled = 0;
+    Enabled = 1;
+    Reset = 2;
+}
+
+enum SandboxKind_XLGpp{
+    SandboxKind_None = 0;
+    Default = 1;
+    WinDetours = 2;
+    MaxOsKext = 3;
+    MacOsKextIgnoreFileAccesses = 4;
+}
+
+enum DoubleWritePolicy_XLGpp{
+    DoubleWritesAreErrors = 0;
+    AllowSameContentDoubleWrites = 1;
+    UnsafeFirstDoubleWriteWins = 2;
+}
+
+message AbsolutePath_XLGpp{
+    string Value = 1;
+}
+
+message ContentHash_XLGpp{
+    string Value = 1;
+}
+
+message FileArtifact_XLGpp{
+    AbsolutePath_XLGpp path = 1;
+
+    int32 RewriteCount = 2;
+
+    bool IsSourceFile = 3; 
+
+    bool IsOutputFile = 4;
+}
+
+message FileContentInfo_XLGpp{
+    // [IsKnownExistenceFlag | Existence | IsKnownLengthFlag | Length]; 

```

Comment:
>// [IsKnownExistenceFlag | Existence | IsKnownLengthFlag | Length];  [](start = 4, length = 68)

should we unfold this? it might be cause lots of headache for consumers of that data otherwise

Reply:
There is no notion of "byte" or splitting the byte into multiple data in protobuf. So we if we decide to split this, it will cost us at least 8 bytes or more. 

---
In reply to: [308894563](https://github.com/microsoft/BuildXL/pull/671#discussion_r308894563) [](ancestors = 308894563)